dynawo/sources/Models/Modelica/Dynawo/Electrical/Shunt.mo

```diff
@@ -0,0 +1,43 @@
+within Dynawo.Electrical;
+
+/*
+* Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model Shunt "Shunt element with voltage dependent reactive power, constant susceptance"
+
+public
+  Connectors.ACPower terminal(V(re(start = u0Pu.re), im(start = u0Pu.im)), i(re(start = i0Pu.re), im(start = i0Pu.im))) "Connector used to connect the shunt to the grid" annotation(
+  Placement(visible = true, transformation(origin = {-1.42109e-14, 98}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-1.42109e-14, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Types.VoltageModulePu UPu(start = ComplexMath.'abs'(u0Pu)) "Voltage amplitude at shunt terminal in p.u (base UNom)";
+  Types.ActivePowerPu PPu(start = s0Pu.re) "Active power at shunt terminal in p.u (base SnRef, receptor convention)";
+  Types.ReactivePowerPu QPu(start = s0Pu.im) "Reactive power at shunt terminal in p.u (base SnRef, receptor convention)";
+  Types.ComplexApparentPowerPu SPu(re (start = s0Pu.re), im (start = s0Pu.im)) "Apparent power at shunt terminal in p.u (base SnRef, receptor convention)";
+
+  parameter Types.ReactivePowerPu QRefPu "Reactive power request at nominal voltage (base SnRef, receptor convention, negative values for capacitive reception, overexcited operation)";  
+
+protected
+  parameter Types.VoltageModulePu UNomPu = 1 "Nominal voltage amplitude";
+  parameter Types.ComplexVoltagePu u0Pu  "Start value of complex voltage at shunt terminal in p.u (base UNom)";
+  parameter Types.ComplexApparentPowerPu s0Pu  "Start value of apparent power at shunt terminal in p.u (base SnRef, receptor convention)";
+  parameter Types.ComplexCurrentPu i0Pu  "Start value of complex current at shunt terminal in p.u (base UNom, SnRef, receptor convention)";
+
+equation
+  SPu = Complex(PPu, QPu);
+  SPu = terminal.V * ComplexMath.conj(terminal.i);     
+  UPu = ComplexMath.'abs'(terminal.V);
+  QPu = QRefPu * (UPu/UNomPu)^2;  
```

Comment:
Is it really UNomPu or U0? For the LoadALphaBeta we have used U0. With U0 we would initially have QPu = QRef but maybe it is not what you intended to do..

Reply:
I would also have naturally gone for U0 for the same reason than @MarianneSaugierRTE

Reply:
I was also thinking to take u0Pu (=U0??) as it is done in LoadAlphaBeta. But according to the report the reactive power of the shunt shall be given for a voltage at 1p.u. fix and u0Pu might deviate from 1p.u. (page 18). Shall I anyway change to u0Pu?

Reply:
Then no, you should probably keep your implementation in order to minimize the source of difference with the reference report :)

Reply:
I would say that in the generic model of a shunt, I'm expected to have something like: Q = Q0  * (U/U0)².
Thus to be consistent and also to be able to handle the report case later on, I suggest the following modifications:

- Add two additional final parameters: Q0Pu and U0Pu which will be defined in the following way:
 `final parameter Types.ReactivePowerPu Q0Pu = ComplexMath.imag(s0Pu) "";'
 ' final parameter Types.VoltageModulePu U0Pu = ComplexMath.'abs'(u0Pu) "";'

- Modifying the QPu equation to 'QPu = Q0Pu * (UPu / U0)²' so you keep a consistent equation (using initial values instead of references ones)

- Delete the QRefPu and UNomPu parameters.

Once you have this, you need to do a bit of calculations to initialize the Nordic test system values for the shunt. Indeed, as you mentioned, the value provided into the report is for 1 p.u. and the load flow calculations give p.u. voltages different to 1 on the connection point for shunts. In our case, what would be required by the model would be u0Pu, i0Pu, and s0Pu. As you know u0Pu from the report and QRef (Q for U = 1 p.u.), you can recalculate the Q0 easily: Q0Pu = s0Pu.im = U0Pu² * QRefPu with QRefPu = QNom from the report.
Using the aforementioned calculation you will end up with Q = QRef for U = UNom but you keep a more generic model in the library and only play with the initial conditions to match the report values.

Does it seem fine for you?

Reply:
@MarianneSaugierRTE what do you think of my proposal? I think it's more in phase with what we do in the other components. It is just a question of available data and way to initialize the model.

Reply:
Yes I like it very much, you are right that the conversion can be made outside the model to keep it consistent with the rest of the library. Thank you for sharing this idea.

Reply:
I also like your idea @adriguir and I am going to implement it that way!